### PR TITLE
Make the `Keyboard Shortcuts` dialog into a settings tab

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: enhancement
Fixes https://github.com/vector-im/element-web/issues/19866

<hr>

---
Here's what your changelog entry will look like:

## ✨ Features
 * Make the `Keyboard Shortcuts` dialog into a settings tab ([\#7198](https://github.com/matrix-org/matrix-react-sdk/pull/7198)). Fixes vector-im/element-web#19866. Contributed by @SimonBrandner.

Preview: https://61e557e2d09d8096b228cef5--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/dialogs/UserSettingsDialog.tsx [hunk 1]
```diff
@@ -118,6 +120,12 @@ export default class UserSettingsDialog extends React.Component<IProps, IState>
             "mx_UserSettingsDialog_preferencesIcon",
             <PreferencesUserSettingsTab closeSettingsFn={this.props.onFinished} />,
         ));
+        tabs.push(new Tab(
+            UserTab.Keyboard,
+            _td("Keyboard"),
+            "mx_UserSettingsDialog_preferencesIcon",
```
Comment (t3chguy) [inline src/components/views/dialogs/UserSettingsDialog.tsx:126]: Needs icon from design

File: src/components/structures/LoggedInView.tsx [hunk 2]
```diff
@@ -502,7 +503,10 @@ class LoggedInView extends React.Component<IProps, IState> {
                 handled = true;
                 break;
             case NavigationAction.ToggleShortCutDialog:
-                KeyboardShortcuts.toggleDialog();
+                dis.dispatch<OpenToTabPayload>({
+                    action: Action.ViewUserSettings,
+                    initialTabId: UserTab.Keyboard,
+                });
```
Comment (t3chguy) [inline src/components/structures/LoggedInView.tsx:509]: The shortcut is called *toggle* but no longer seems to toggle

Comment (SimonBrandner) [reply src/components/structures/LoggedInView.tsx:509]: Yeah, though I am wondering if it should be that way, now that it is a settings tab, will leave it up to design to choose what is ideal here

Comment (t3chguy) [reply src/components/structures/LoggedInView.tsx:509]: Needs input from @matrix-org/design

Comment (t3chguy) [reply src/components/structures/LoggedInView.tsx:498]: I think just changing it to `OpenShortcutDialog` is sane (and the accompanying copy also)

**Review by t3chguy — COMMENTED** [review]
I think the `Keyboard Shortcuts` button in Settings > Help & About is now redundant, though maybe the link in Preferences is sane for now.

**Review by t3chguy — CHANGES_REQUESTED** [review]
The dialog's own shortcut still says "Toggle this dialog"

**Review by t3chguy — APPROVED** [review]

**Review by amshakal — APPROVED** [review]

**Comment by t3chguy** [comment]
@SimonBrandner whats blocking this? also could you merge in latest develop please, hopefully will unwedge the test

**Comment by SimonBrandner** [comment]
> @SimonBrandner whats blocking this? 

We still don't have the icon, @amshakal said they'd look into it

**Comment by t3chguy** [comment]
@amshakal what's the status here?

**Comment by amshakal** [comment]
Was given this request an hour back, I'll share an icon by EOD. Thanks

**Comment by amshakal** [comment]
Link to keyboard icon: https://www.figma.com/file/X4XTH9iS2KGJ2wFKDqkyed/Compound?node-id=1373%3A13

**Pull request merged**